docs/data/date-pickers/date-picker/JalaliDatePicker.tsx

```diff
@@ -1,14 +1,14 @@
 import * as React from 'react';
 import TextField from '@mui/material/TextField';
-import AdapterJalali from '@date-io/date-fns-jalali';
+import { AdapterDateFnsJalali } from '@mui/x-date-pickers/AdapterDateFnsJalali';
 import { DatePicker } from '@mui/x-date-pickers/DatePicker';
 import { LocalizationProvider } from '@mui/x-date-pickers/LocalizationProvider';
 
 export default function JalaliDatePicker() {
   const [value, setValue] = React.useState<Date | null>(new Date(2022, 3, 7));
 
   return (
-    <LocalizationProvider dateAdapter={AdapterJalali}>
+    <LocalizationProvider dateAdapter={AdapterDateFnsJalali}>
```

Comment:
I guess this is not related to this PR
Maybe we could extract it into a standalone PR

But this section will most likely go away after #6625 anyway

Same for the change in the `.md` file

Reply:
Ok, I first taught I mixed two PR, but in fact it's not the case.

Now that we have all the adapter exported internally (except js-joda) I added the `getWeekNumber` method to all the adapters. THen I realized the doc was still encouraging to import them from date-io so I made a correction